java/flight/flight-core/src/main/java/org/apache/arrow/flight/SetSessionOptionsRequest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SetSessionOptionsRequest {
+  private final Map<String, SessionOptionValue> session_options;
+
+  public SetSessionOptionsRequest(Map<String, SessionOptionValue> session_options) {
+    this.session_options = ConcurrentHashMap(session_options);
```

Comment:
new ConcurrentHashMap.

Reply:
Why use a ConcurrentHashMap? It doesn't look like the map is even mutable after being passed to the constructor?